### PR TITLE
Get pseudonym

### PR DESCRIPTION
**Pull request by tpazderka**

### CODE REVIEW

File: cz_nia/__init__.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
 """Python application for communication with Czech NIA."""
 
 __version__ = '0.0'
+
+
+class NIAException(Exception):
+    """Custom class for NIA exceptions."""
```
Comment (ziima) [inline cz_nia/__init__.py]: I'd move this to the `exceptions` submodule.

File: cz_nia/functions.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    SUBMISSION = 'http://www.government-gateway.cz/wcf/submission'
+
+
+@unique
+class NIAConstants(str, Enum):
+    """Enum for NIA constants."""
+
+    ASSERTION = 'urn:oasis:names:tc:SAML:1.0:assertion'
+    WS_TRUST_ISSUE = 'http://docs.oasis-open.org/ws-sx/ws-trust/200512/Issue'
+    WS_TRUST_SYM_KEY = 'http://docs.oasis-open.org/ws-sx/ws-trust/200512/SymmetricKey'
```
Comment (ziima) [inline cz_nia/functions.py]: These doesn't look like a good use cases for enums.

File: cz_nia/functions.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    service = client.bind('SecurityTokenService', 'WS2007HttpBinding_IWSTrust13Sync2')
+    try:
+        response = service.Trust13Issue(_value_1=[token, request, key, applies])
+    except Error as err:
+        raise NIAException(err)
+    return response.RequestSecurityTokenResponse[0]['_value_1'][3]['_value_1']
+
+
+def _call_fpsts(settings, transport, assertion):
+    """Call FPSTS service and return the assertion."""
```
Comment (ziima) [inline cz_nia/functions.py]: I'd try to translate `ipsts` and `fpsts` into some legible names if possible.

File: cz_nia/message.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        compare_type = SubElement(id_request, QName(self.request_namespace, 'TypPorovnani'))
+        compare_type.text = 'diakritika'
+        return id_request
+
+    def extract_message(self, response):
+        """Get pseudonym from the message."""
+        return response.find('nia:Pseudonym', namespaces=self.get_namespace_map).text
+
+
+NIAMessage.register(ZtotozneniMessage)
```
Comment (ziima) [inline cz_nia/message.py]: This line doesn't seem to be necessary.

File: cz_nia/settings.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+"""CZ_NIA applocation settings wrapper."""
+from __future__ import unicode_literals
+
+
+class CzNIAAppSettings(object):
```
Comment (ziima) [inline cz_nia/settings.py]: I'd use `Nia` in class names.

**Comment by tpazderka** [comment]
Updated.

File: cz_nia/settings.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+"""CZ_NIA applocation settings wrapper."""
```
Comment (ziima) [inline cz_nia/settings.py]: `application`

File: cz_nia/message.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        name.text = self.data.get('first_name')
+        surname = SubElement(id_request, QName(self.request_namespace, 'Prijmeni'))
+        surname.text = self.data.get('last_name')
+        date_of_birth = SubElement(id_request, QName(self.request_namespace, 'DatumNarozeni'))
+        date_of_birth.text = self.data.get('birth_date').isoformat()
+        compare_type = SubElement(id_request, QName(self.request_namespace, 'TypPorovnani'))
+        compare_type.text = 'diakritika'
+        return id_request
+
+    def extract_message(self, response):
```
Comment (ziima) [inline cz_nia/message.py]: This class is missing annotations.

File: cz_nia/tests/data/FPSTS_nice.wsdl [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        <!--VZIMA: Switch choices to sequence -->
+        <!--
+        <xs:choice minOccurs="0" maxOccurs="unbounded">
+          <xs:any minOccurs="0" maxOccurs="unbounded" namespace="##any" processContents="lax"/>
+        </xs:choice>
+        -->
+        <xs:sequence>
+          <xs:any minOccurs="0" maxOccurs="unbounded" namespace="##any" processContents="lax"/>
+        </xs:sequence>
+        <!--VZIMA: End -->
```
Comment (ziima) [inline cz_nia/tests/data/FPSTS_nice.wsdl]: I'd fix the comments.

File: cz_nia/tests/data/fp_token.xml [hunk 9]
```diff
@@ -0,0 +1 @@
...AgIAgDAHBgUrDgMCBzAKBggqhkiG9w0DBzALBglghkgBZQMEASowCwYJYIZIAWUDBAEtMAsGCWCGSAFlAwQBFjALBglghkgBZQMEARkwCwYJYIZIAWUDBAECMAsGCWCGSAFlAwQBBTAdBgNVHQ4EFgQUpnwhDx+6TFt7x+EtlQSi0BNFy9IwHwYDVR0jBBgwFoAUrdPvn/QyVybg3jQkzBkb08NDx3wwgc8GA1UdHwSBxzCBxDCBwaCBvqCBu4aBuGxkYXA6Ly8vQ049VEdHLUNBLU1PUklTLENOPVRHRy1DQS0wMSxDTj1DRFAsQ049UHVibGljJTIwS2V5JTIwU2VydmljZXMsQ049U2VydmljZXMsQ049Q29uZmlndXJhdGlvbixEQz10Z2dtb3JpcyxEQz1sb2NhbD9jZXJ0aWZpY2F0ZVJldm9jYXRpb25MaXN0P2Jhc2U/b2JqZWN0Q2xhc3M9Y1JMRGlzdHJpYnV0aW9uUG9pbnQwgcEGCCsGAQUFBwEBBIG0MIGxMIGuBggrBgEFBQcwAoaBoWxkYXA6Ly8vQ049VEdHLUNBLU1PUklTLENOPUFJQSxDTj1QdWJsaWMlMjBLZXklMjBTZXJ2aWNlcyxDTj1TZXJ2aWNlcyxDTj1Db25maWd1cmF0aW9uLERDPXRnZ21vcmlzLERDPWxvY2FsP2NBQ2VydGlmaWNhdGU/YmFzZT9vYmplY3RDbGFzcz1jZXJ0aWZpY2F0aW9uQXV0aG9yaXR5MA0GCSqGSIb3DQEBBQUAA4IBAQAfzNajiNN1RFhmpDT4sBjKt/jN3Q4yc/rrXW8DJUtVoFZXKYZ5sDepLK8w7KOm1KgERvWv/XzxG8+iM2gCMmlhimUFPxCKLUICbqwjJW+khsLfNxqgA6oAc0qqRvi3UhaHxdViFhwZXuU6pK7+o6SLOjO6OKB4wqfr1nZ3esmhnRBqvOzSbhWBBYArwJ9IJGIK+mHvnlpGOFpFDr67z/hl5JsPphJyB90s7LDdGfugBEPSgZWHp4jE0fOpJKl/S8gq9E4jo0oFrqisnf1f16F3Am4SUS+dTVgSnQKzxgiCwx48DhCQ9TG2f7Yz1Q1ckPF8JBUtkbPzXHAUvgQNzMVi</X509Certificate></X509Data></KeyInfo></Signature></saml:Assertion>
```
Comment (ziima) [inline cz_nia/tests/data/fp_token.xml]: I'd make the XML a bit prettier.

**Comment by tpazderka** [comment]
Updated

**Pull request merged**